File: locust/ve.py
from locust import HttpLocust, TaskSet, task
from datetime import datetime
import json


class VeTaskSet(TaskSet):

    @task
    def make(self):
        data = {
            "tplId": 1,
            "imgUrls": "http://t-ss2.meipian.me/config/1545803143072.png",
            "tplVersion": 1,
            "tplUrl": "https://static-musiccard.ivwen.com/tplmusic/1.zip",
            "recordId": "2019012810055519",
            "createdAt": str(datetime.timestamp),
            "uploadKey": "2019012810055519.mp4"
        }
        headers = {"Content-Type": "application/json"}
        
        with self.client.post('/task/commit', headers=headers, data=json.dumps(data)) as response:
            # result = response.json()
            print(response.raise_for_status())


class VeLocust(HttpLocust):
    host = 'http://172.17.31.157'
    task_set = VeTaskSet
    min_wait = 0
    max_wait = 0

File: wrk/ve.lua

--  wrk -c1 -d1s -t1 --script=ve.lua http://172.17.31.157/task/commit
start_time = os.time()

function request()
    -- body
    local headers = {}
    headers["Content-Type"] = "application/json"
    local body = {}
    body["tplId"] = 1
    body["imgUrls"] = "http://t-ss2.meipian.me/config/1545803143072.png"
    body["tplVersion"] = 1
    body["tplUrl"] = "https://static-musiccard.ivwen.com/tplmusic/1.zip"
    body["recordId"] = "2019012810055519"
    body["createdAt"] = os.time()
    body["uploadKey"] = "2019012810055519.mp4"
    -- body = "user_id=30402206&template_id=1&material=http://t-ss2.meipian.me/config/1545803143072.png"
    local cjson = require("cjson")
    body_str = cjson.encode(body)
    print(body_str)
    return wrk.format("POST", nil, headers, body_str)
end

function response(status, headers, body)
    -- body
    print(body)
end

function done(summary, latency, requests)
    -- body
    end_time = os.time()
    diff_time = os.difftime(end_time, start_time)
    print("总共耗时:"..diff_time)
end

